games-strategy/ja2-stracciatella/ja2-stracciatella-0.17.0.ebuild

```diff
@@ -0,0 +1,188 @@
+md-5-0.8.0
+memchr-2.3.3
+memoffset-0.5.4
+num-integer-0.1.42
+num-traits-0.2.11
+num_cpus-1.12.0
+opaque-debug-0.2.3
+ppv-lite86-0.2.6
+proc-macro2-1.0.10
+quote-1.0.3
+rand-0.7.3
+rand_chacha-0.2.2
+rand_core-0.5.1
+rand_hc-0.2.0
+rayon-1.3.0
+rayon-core-1.7.0
+redox_syscall-0.1.56
+redox_users-0.3.4
+regex-1.3.6
+regex-syntax-0.6.17
+remove_dir_all-0.5.2
+rust-argon2-0.7.0
+ryu-1.0.3
+scopeguard-1.1.0
+serde-1.0.105
+serde_derive-1.0.105
+serde_json-1.0.50
+simplelog-0.6.0
+smallvec-1.2.0
+strsim-0.8.0
+syn-1.0.17
+tempfile-3.1.0
+term-0.5.2
+textwrap-0.11.0
+thread_local-1.0.1
+time-0.1.42
+toml-0.5.6
+typenum-1.11.2
+unicode-normalization-0.1.12
+unicode-width-0.1.7
+unicode-xid-0.2.0
+vec_map-0.8.1
+wasi-0.9.0+wasi-snapshot-preview1
+winapi-0.3.8
+winapi-i686-pc-windows-gnu-0.4.0
+winapi-x86_64-pc-windows-gnu-0.4.0
+"
+
+inherit cargo cmake xdg-utils
+
+DESCRIPTION="An improved, cross-platform, stable Jagged Alliance 2 runtime"
+HOMEPAGE="https://github.com/ja2-stracciatella/"
+
+SRC_URI="https://github.com/ja2-stracciatella/ja2-stracciatella/releases/tag/v${PV}.tar.gz -> ${P}.tar.gz
+	editor? ( https://github.com/ja2-stracciatella/free-ja2-resources/releases/download/v1/editor.slf -> ${P}-editor.slf )
+"
+SRC_URI+=" $(cargo_crate_uris ${CRATES})"
+
+IUSE="cdinstall editor ru-gold test"
+RESTRICT="test"
+LICENSE=" CC-BY-NC-2.0 SFI-SCLA "
+SLOT="0"
+KEYWORDS="~amd64"
+
+DEPEND="media-libs/libsdl2[X,sound,video]
+		!~media-libs/libsdl2-2.0.6
+		>=dev-lang/rust-1.40.0
+		>=x11-libs/fltk-1.3.5[opengl]
+		>=dev-cpp/string-theory-3.1
+		>=dev-games/libsmacker-1.1.1
+		>=dev-libs/rapidjson-1.1.0
+"
+
+BDEPEND="
+		test? ( >=dev-cpp/gtest-1.9.0_pre20190607 )
+"
+
+RDEPEND="${DEPEND}
+	cdinstall? ( games-strategy/ja2-stracciatella-data )
+"
+
+LANGS="l10n_de +l10n_en l10n_fr l10n_it l10n_nl l10n_pl l10n_ru"
+
+IUSE="$IUSE $LANGS"
+REQUIRED_USE="^^ ( ${LANGS//+/} )"
+
+src_configure() {
+		local mycmakeargs=()
+
+		case ${L10N} in
+			de) mycmakeargs+=" -DLNG=GERMAN" ;;
+			nl) mycmakeargs+=" -DLNG=DUTCH" ;;
+			fr) mycmakeargs+=" -DLNG=FRENCH" ;;
+			it) mycmakeargs+=" -DLNG=ITALIAN" ;;
+			pl) mycmakeargs+=" -DLNG=POLISH" ;;
+			ru) mycmakeargs+=" -DLNG=$(usex ru-gold RUSSIAN_GOLD RUSSIAN)" ;;
+			en) mycmakeargs+=" -DLNG=ENGLISH" ;;
+			*) die "no language selected in L10N" ;;
+		esac
+		elog "Chosen language is ${mycmakeargs# -DLNG=}"
+
+		mycmakeargs+=(
+			-DLOCAL_GTEST_LIB=OFF
+			-DLOCAL_RAPIDJSON_LIB=OFF
+			-DLOCAL_STRING_THEORY_LIB=OFF
+			-DWITH_UNITTESTS=$(usex test)
+			-DWITH_RUST_BINARIES=OFF
+			-DBUILD_LAUNCHER=OFF
+			-DOPENGL_GL_PREFERENCE=GLVND
+			-DEXTRA_DATA_DIR="${EPREFIX}/usr/share/ja2"
+		)
+
+		cargo_gen_config
+		cmake_src_configure
+}
+
+src_install() {
+	if use editor; then
+		insinto /usr/share/ja2
+		doins "${DISTDIR}/${P}-editor.slf"
+		dosym "${P}-editor.slf" "/usr/share/ja2/editor.slf"
+	fi
+
+	cmake_src_install
+}
+
+pkg_postinst() {
+	elog "You need ja2 in the chosen language, otherwise set it in package.use!"
```

Comment:
There's more than one way to set a USE variable.

Reply:
You need to specify exactly ONE language for the DATA CD.

Reply:
This sounds like a REQUIRED_USE job?

Reply:
I don`t know.. I just took the ebuild which is over a decade old now and it was always like that.

Reply:
REQUIRED_USE is set already above